libcxx/include/__memory/allocator.h

```diff
@@ -76,8 +79,10 @@ struct __non_trivial_if<true, _Unique> {
 
 template <class _Tp>
 class _LIBCPP_TEMPLATE_VIS allocator : private __non_trivial_if<!is_void<_Tp>::value, allocator<_Tp> > {
-  static_assert(!is_const<_Tp>::value, "std::allocator does not support const types");
-  static_assert(!is_volatile<_Tp>::value, "std::allocator does not support volatile types");
+  static_assert(!is_const<_Tp>::value, "'std::allocator' cannot allocate const types");
+  static_assert(!is_volatile<_Tp>::value, "'std::allocator' cannot allocate volatile types");
+  static_assert(!is_reference<_Tp>::value, "'std::allocator' cannot allocate references");
+  static_assert(!is_function<_Tp>::value, "'std::allocator' cannot allocate functions");
```

Comment:
Can we tie these checks back to the Standard wording that says so? It doesn't have to be in the static assert message, but at least in a comment.

Reply:
PTAL